src/compiler/program.ts

```diff
@@ -1290,12 +1287,16 @@ namespace ts {
                 }
             }
 
-            function collectRequireCalls(node: Node): void {
+            function collectDynamicImportOrRequireCalls(node: Node): void {

```

Comment:
Yea. Incremental parser is the main reason I pick this approach. We can try to be smarter about this but it will add lots of complexity to get our incremental parser...I do think it will be worth visiting this again in the future ... However with the new NodeFlags.possiblyContainDynamicImport  at least people who don't use the feature won't pay the price